shopping-cart/shopping-cart-scala/shopping-cart/src/main/scala/com/example/shoppingcart/impl/LagomTaggerAdapter.scala

```diff
@@ -0,0 +1,25 @@
+package com.example.shoppingcart.impl
+
+import com.lightbend.lagom.scaladsl.persistence.AggregateEventTag
+import com.lightbend.lagom.scaladsl.persistence.AggregateEventTagger
+import com.lightbend.lagom.scaladsl.persistence.AggregateEventShards
+import com.lightbend.lagom.scaladsl.persistence.AggregateEvent
+import akka.cluster.sharding.typed.scaladsl.EntityContext
+
+object LagomTaggerAdapter {
+
+  
+  def apply[Event <: AggregateEvent[Event]](
+      entityCtx: EntityContext,
+      lagomTagger: AggregateEventTagger[Event]
+  ): Event => Set[String] = { evt =>
+    val tag =
+      lagomTagger match {
+        case tagger: AggregateEventTag[_] =>
+          tagger.tag
+        case shardedTagger: AggregateEventShards[_] =>
+          shardedTagger.forEntityId(entityCtx.entityId).tag
+      }
+    Set(tag)
+  }
+}
```

Comment:
This needs to go into Lagom. Is an adapter for a AggregateEvent to Akka's new tagging function (ie: `Event => Set[String]`).

Note that I'm requesting the `EntityContext` to be passed and taking control over which id to use to partition the tags.
The reason being that Lagom will use the [business id to partition the tags](https://github.com/lagom/lagom/blob/master/persistence/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/PersistentEntityActor.scala#L242), not the [id with the type hint](https://github.com/lagom/lagom/blob/master/persistence/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/PersistentEntityActor.scala#L104). 

I'm happy that we are [removing the `EventSourcedEntity`](https://github.com/akka/akka/pull/27725) from Akka, because that would have given us only the id with the type hint causing the partition numbers to be re-defined.

@patriknw ☝️

Reply:
Yes, let's do like that when moving this into Lagom.